src/smbprotocol/exceptions.py

```diff
@@ -47,7 +49,7 @@ class SMBOSError(OSError, SMBException):
 
     def __init__(self, ntstatus, filename, filename2=None):
         self.ntstatus = ntstatus
-        self.filename2 = str(filename2) if filename2 else None
+        self.filename2 = os.fspath(filename2) if filename2 else None
```

Comment:
https://docs.python.org/3/library/os.html#os.fspath

It is recommended to convert file paths matching `os.PathLike` protocol (like `pathlib.Path`) to strings instead of using just `str(...)`

Reply:
Thanks for the info, makes sense.